plutus-scb-client/src/MainFrame.purs

```diff
@@ -100,8 +117,51 @@ handleAction (ChainAction newFocus) = do
     (_chainState <<< _chainFocusAppearing)
     (zoomStateT _chainState $ Chain.handleAction newFocus mAnnotatedBlockchain)
 
-handleAction (InvokeContractEndpoint subaction) = do
-  liftEffect $ log $ "TRIGGER: " <> show subaction
+handleAction (ChangeContractEndpointCall contractInstanceId endpointIndex argumentIndex subaction) = do
+  modifying
+    ( _contractSignatures
+        <<< ix contractInstanceId
+        <<< _Success
+        <<< ix endpointIndex
+        <<< _arguments
+        <<< ix argumentIndex
+    )
+    (Schema.handleFormEvent initialValue subaction)
+
+handleAction (InvokeContractEndpoint contractInstanceId endpointForm) = do
+  let
+    endpointDescription :: EndpointDescription
+    endpointDescription = view (_schema <<< _FunctionSchema <<< _endpointDescription) endpointForm
+
+    encodedForm :: Warn (Text "It shouldn't be possible to get anything other than exactly 1 argument, but the code implies otherwise.") => Maybe RawJson
+    encodedForm = do
+      argument1 <- Array.head $ view _arguments endpointForm
+      pure $ RawJson $ encodeJSON $ formArgumentToJson argument1
+  for_ encodedForm
+    $ \argument -> do
+        result <-
+          runAjax
+            $ putContractByContractinstanceidEndpointByEndpointname argument
```

Comment:
wait, am I looking at Java or purescript?

Reply:
Ha. This is an autogenerated API name from servant-purescript. So it is what it is. 🤷‍♂️